dirsrvtests/tests/suites/plugins/attruniq_test.py

```diff
@@ -154,3 +168,158 @@ def test_multiple_attr_uniqueness(topology_st):
+    user4 = users_excluded2.create(properties={'cn': USER_4_CN,
+                                               'uid': USER_4_CN,
+                                               'sn': USER_4_CN,
+                                               'uidNumber': '4',
+                                               'gidNumber': '44',
+                                               'homeDirectory': '/home/{}'.format(USER_4_CN)})
+
+    UNIQUE_VALUE = '1234'
+
+    log.info('Create user with unique attribute')
+    user1.add('telephonenumber', UNIQUE_VALUE)
+    assert user1.present('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Add exclude subtree')
+    attruniq.add_exclude_subtree(EXCLUDED_CONTAINER_DN)
+    topology_st.standalone.restart()
+
+    log.info('Verify an already used attribute value cannot be added within the same subtree')
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        user2.add('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Verify an entry with same attribute value can be added within exclude subtree')
+    user3.add('telephonenumber', UNIQUE_VALUE)
+    assert user3.present('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Cleanup unique attribute values')
+    user1.remove_all('telephonenumber')
+    user3.remove_all('telephonenumber')
+
+    log.info('Add a unique value to an entry in excluded scope')
+    user3.add('telephonenumber', UNIQUE_VALUE)
+    assert user3.present('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Verify the same value can be added to an entry within uniqueness scope')
+    user1.add('telephonenumber', UNIQUE_VALUE)
+    assert user1.present('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Verify that yet another same value cannot be added to another entry within uniqueness scope')
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        user2.add('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Cleanup unique attribute values')
+    user1.remove_all('telephonenumber')
+    user3.remove_all('telephonenumber')
+
+    log.info('Add another exclude subtree')
+    attruniq.add_exclude_subtree(EXCLUDED_BIS_CONTAINER_DN)
+    topology_st.standalone.restart()
+
+    user1.add('telephonenumber', UNIQUE_VALUE)
+    log.info('Verify an already used attribute value cannot be added within the same subtree')
+    with pytest.raises(ldap.CONSTRAINT_VIOLATION):
+        user2.add('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Verify an already used attribute can be added to an entry in exclude scope')
+    user3.add('telephonenumber', UNIQUE_VALUE)
+    assert user3.present('telephonenumber', UNIQUE_VALUE)
+    user4.add('telephonenumber', UNIQUE_VALUE)
+    assert user4.present('telephonenumber', UNIQUE_VALUE)
+
+    log.info('Clean up users')
+    user1.delete()
+    user2.delete()
+    user3.delete()
+    user4.delete()
+    cont1.delete()
+    cont2.delete()
+    cont3.delete()
+    attruniq.disable()
+    attruniq.delete()
```

Comment:
The deletes, probably, can be done in a loop.

Also, just to be safe, I'd do the clean up in the `finally` block (or any other similar approach)